File: GamesTournamentsWeb.Web/ClientApp/models/tournaments/TournamentFilter.ts
import { ConvertableToJson } from '~/models/ConvertableToJson'

class TournamentFilter extends ConvertableToJson {
  public name: string | null
  public gameId: number | null
  public teamSizes: number[]
  public regionIds: number[]
  public platformIds: number[]
  public genreIds: number[]
  public withMyTournaments: boolean

  constructor () {
    super()

    this.name = null
    this.gameId = null
    this.teamSizes = []
    this.regionIds = []
    this.platformIds = []
    this.genreIds = []
    this.withMyTournaments = false
  }
}

export { TournamentFilter }

File: GamesTournamentsWeb.Infrastructure/Dto/Tournaments/TournamentFilter.cs
namespace GamesTournamentsWeb.Infrastructure.Dto.Tournaments;

public class TournamentFilter
{
    public string Name { get; set; }
    public int? GameId { get; set; }
    public int[] TeamSizes { get; set; }
    public int[] RegionIds { get; set; }
    public int[] PlatformIds { get; set; }
    public int[] GenreIds { get; set; }
    public bool WithMyTournaments { get; set; }
    public int Page { get; set; }
}